### PR TITLE
Pass through solc opts

### PR DESCRIPTION
**Pull request by HenryNguyen5**

### CODE REVIEW

**Comment by samsondav** [comment]
Why?

File: core/services/vrf/vrf_fulfillment_cost_test.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ func TestMeasureFulfillmenttGasCost(t *testing.T) {
 		coordinator.rootContractAddress, coordinator.coordinatorABI,
 		"fulfillRandomnessRequest", proofBlob[:])
 
-	assert.Greater(t, estimate, uint64(148000),
+	assert.Greater(t, estimate, uint64(145000),
```
Comment (samsondav) [inline core/services/vrf/vrf_fulfillment_cost_test.go:25]: Why this change?

File: evm-contracts/app.config.json [hunk 2]
```diff
@@ -4,6 +4,10 @@
   "contractAbstractionDir": ".",
   "useDockerisedSolc": false,
   "compilerSettings": {
+    "optimizer": {
```
Comment (samsondav) [inline evm-contracts/app.config.json:7]: What does this do?

**Comment by kaleofduty** [comment]
> What is this about?

Fixing a bug in our configuration handling that caused the solidity optimizer to not run, even when it was enabled  in `evm-contracts/app.config.json`

**Comment by kaleofduty** [comment]
Does this mean that we will now generate different bytecode for the already deployed contracts?

File: core/services/vrf/vrf_fulfillment_cost_test.go (hunk 1 above)
Comment (se3000) [reply core/services/vrf/vrf_fulfillment_cost_test.go:25]: Optimization lowered the best case for gas usage.

**Comment by se3000** [comment]
Bytecode changing will depend on how we deploy contracts. Right now we use tools outside the repo, so they will continue to match. I think @HenryNguyen5's vision of `belt` is that it would turn into a tool belt that we could use for things like command line deployments. I'm in support of that, if only so that our tests can be a little closer to our actual deployments.

One interesting thing about our current deployments not matching our tests: as the FluxAggregator grows it will require the optimizer to be enabled to fit the deployment limits.

File: evm-contracts/app.config.json (hunk 2 above)
Comment (se3000) [reply evm-contracts/app.config.json:7]: This is the core of the change. These options weren't getting passed along, but we save a lot of space in the size of the compiled contract by with this configuration. (~40% savings based on the contract I was testing with.)
